crates/common/src/raindex_client/mod.rs

```diff
@@ -98,60 +111,85 @@ impl RaindexClient {
+                    let orderbooks = self
+                        .orderbook_yaml
+                        .get_orderbooks_by_network_key(&network.key)?;
+                    for orderbook in orderbooks {
+                        multi_subgraph_args.entry(id).or_insert(Vec::new()).push(
+                            MultiSubgraphArgs {
+                                url: orderbook.subgraph.url.clone(),
+                                name: network.label.clone().unwrap_or(network.key.clone()),
+                            },
+                        );
+                    }
+                }
+                multi_subgraph_args
             }
-            None => {
-                let mut multi_subgraph_args = HashMap::new();
+            Some(_) | None => {
+                let mut multi_subgraph_args = BTreeMap::new();
                 let networks = self.orderbook_yaml.get_networks()?;
 
                 for network in networks.values() {
-                    let orderbook = self
+                    let orderbooks = self
                         .orderbook_yaml
-                        .get_orderbook_by_network_key(&network.key)?;
-                    multi_subgraph_args.insert(
-                        network.chain_id,
-                        MultiSubgraphArgs {
-                            url: orderbook.subgraph.url.clone(),
-                            name: network.label.clone().unwrap_or(network.key.clone()),
-                        },
-                    );
-                }
-
-                if multi_subgraph_args.is_empty() {
-                    return Err(RaindexError::NoNetworksConfigured);
+                        .get_orderbooks_by_network_key(&network.key)?;
+                    for orderbook in orderbooks {
+                        multi_subgraph_args
+                            .entry(network.chain_id)
+                            .or_insert(Vec::new())
+                            .push(MultiSubgraphArgs {
+                                url: orderbook.subgraph.url.clone(),
+                                name: network.label.clone().unwrap_or(network.key.clone()),
+                            });
+                    }
                 }
-
                 multi_subgraph_args
             }
         };
 
-        Ok(result.into_iter().collect::<BTreeMap<_, _>>())
+        if result.is_empty() {
+            return Err(RaindexError::NoNetworksConfigured);
+        }
+        Ok(result)
+    }
+
+    #[wasm_export(skip)]
+    pub fn get_orderbook_client(
+        &self,
+        chain_id: u32,
+        orderbook_address: Address,
+    ) -> Result<OrderbookSubgraphClient, RaindexError> {
+        let orderbook = self.get_orderbook_for_chain_id(chain_id, orderbook_address)?;
+        Ok(OrderbookSubgraphClient::new(orderbook.subgraph.url.clone()))
     }
 
-    fn get_subgraph_url_for_chain(&self, chain_id: u64) -> Result<Url, RaindexError> {
+    fn get_orderbook_for_chain_id(
```

Comment:
i don't get why you need the chainid if you already have the orderbook address?

Reply:
you are right, that was an artifact from the logic before the new url structure. we can get the orderbook directly from the address